app/models/reports/schema_v2.rb

```diff
@@ -0,0 +1,228 @@
+    end
+
+    # Returns the earliest patient record for a Region from either assigned or registered patients. Note that this *ignores*
+    # the periods that are passed in for the Repository - this is the true 'earliest report date' for a Region.
+    memoize def earliest_patient_recorded_at
+      region_entries = regions.map { |region| RegionEntry.new(region, __method__) }
+      cached_results = cache.fetch_multi(*region_entries, force: bust_cache?) { |region_entry|
+        earliest_patient_data_query_v2(region_entry.region)
+      }
+      cached_results.each_with_object({}) { |(region_entry, time), results| results[region_entry.slug] = time }
+    end
+
+    memoize def earliest_patient_recorded_at_period
+      earliest_patient_recorded_at.each_with_object({}) { |(slug, time), hsh| hsh[slug] = Period.new(value: time, type: @period_type) if time }
+    end
+
+    private def earliest_patient_data_query_v2(region)
+      FacilityState.for_region(region).where("cumulative_registrations > 0 OR cumulative_assigned_patients > 0")
+        .minimum(:month_date)
+    end
+
+    # Adjusted patient counts are the patient counts from three months ago (the adjusted period) that
+    # are the basis for control rates. These counts DO include lost to follow up.
+    memoize def adjusted_patients_with_ltfu
+      cumulative_assigned_patients.each_with_object({}) do |(entry, result), results|
+        values = periods.each_with_object(Hash.new(0)) { |period, region_result|
+          region_result[period] = result[period.adjusted_period]
+        }
+        results[entry] = values
+      end
+    end
+
+    # Adjusted patient counts are the patient counts from three months ago (the adjusted period) that
+    # are the basis for control rates. These counts DO NOT include lost to follow up.
+    memoize def adjusted_patients_without_ltfu
+      regions.each_with_object({}) { |region, result|
+        result[region.slug] = adjusted_patients_without_ltfu_query(region).slice(*periods.entries)
+      }
+    end
+
+    private def adjusted_patients_without_ltfu_query(region)
+      FacilityState.for_region(region)
+        .order(:month_date)
+        .group(:month_date)
+        .sum("patients_under_care::int")
+        .to_h(&period_hash)
+    end
+
+    alias_method :adjusted_patients, :adjusted_patients_without_ltfu
+
+    # Return the running total of cumulative assigned patient counts. Note that this *includes* LTFU.
+    memoize def cumulative_assigned_patients
+      regions.each_with_object({}) { |region, result|
+        result[region.slug] = cumulative_assigned_patients_query_v2(region)
+      }
+    end
+
+    # Returns cumulative assigned patients from facility_states - this includes LTFU
+    private def cumulative_assigned_patients_query_v2(region)
+      FacilityState.for_region(region)
+        .order(:month_date)
+        .group(:month_date)
+        .sum("cumulative_assigned_patients::int")
```

Comment:
We cast to `int` because otherwise the summed values are returned as floats...which doesn't matter much in the Reports (we show integers there)...but does make comparing and debugging in dev/test troublesome.  Hence the cast.

I'm not really sure why PSQL / Rails return the summed values as floats in the first place, but its not a big deal right now.

Reply:
I'll try to fix this in another PR. Created a story to track it here: https://app.clubhouse.io/simpledotorg/story/4311/ensure-integer-values-from-the-schema-are-returned-as-integers-not-floats

Reply:
Neat. This PR doesn't have to wait for that fix either, which is nice. We can rip out the typecasts when ready later.